packages/core/src/api/resolvers/base/base-auth.resolver.ts

```diff
@@ -143,11 +142,13 @@ export class BaseAuthResolver {
     /**
      * Exposes a subset of the User properties which we want to expose to the public API.
      */
-    protected publiclyAccessibleUser(user: User): CurrentUser {
+    protected async publiclyAccessibleUser(user: User): Promise<CurrentUser> {
         return {
             id: user.id,
             identifier: user.identifier,
-            channels: getUserChannelsPermissions(user) as CurrentUserChannel[],
+            channels: (await this.configService.authOptions.rolePermissionResolverStrategy.resolvePermissions(
```

Comment:
When `as` is required I usually assume that this line can be improved

Reply:
Agreed, hate it. Turns out typescript is smart enough to understand the overlap between the types. :bulb:  Removed it now.